restcomm/restcomm.testsuite/src/test/java/org/restcomm/connect/testsuite/sms/push/SmsPushNotificationServerTest.java

```diff
@@ -0,0 +1,247 @@
+
+        Credential c = new Credential("127.0.0.1", "bob", CLIENT_PASSWORD);
+        bobPhone.addUpdateCredential(c);
+        assertTrue(bobPhone.register(uri, "bob", CLIENT_PASSWORD, bobContact, 3600, 3600));
+
+        SipCall bobCall = bobPhone.createSipCall();
+        bobCall.initiateOutgoingMessage(bobContact, aliceContact, null, null, null, "Hello, Alice!");
+        assertTrue(bobCall.waitForAuthorisation(5000));
+        assertTrue(bobCall.waitOutgoingMessageResponse(5000));
+        assertTrue(aliceCall.waitForMessage(5000));
+        assertTrue(aliceCall.sendMessageResponse(Response.ACCEPTED, "Accepted", 3600, null));
+
+        int response;
+        do {
+            assertTrue(bobCall.waitOutgoingCallResponse(5 * 1000));
+            response = bobCall.getLastReceivedResponse().getStatusCode();
+        } while (response == Response.TRYING);
+        assertTrue(response == Response.ACCEPTED);
+
+        verify(postRequestedFor(urlEqualTo("/api/notifications")));
+    }
+
+    @Test
+    public void testSmsEndpointMessage() throws ParseException {
+
+        stubFor(post(urlPathEqualTo("/api/notifications"))
+                .withHeader("Content-Type", matching("application/json;.*"))
+                .willReturn(aResponse()
+                        .withStatus(200)));
+
+        final SipURI uri = aliceSipStack.getAddressFactory().createSipURI(null, restcommContact);
+
+        wireMockRule.addMockServiceRequestListener(new RequestListener() {
+            @Override
+            public void requestReceived(com.github.tomakehurst.wiremock.http.Request request, com.github.tomakehurst.wiremock.http.Response response) {
+                if (request.getAbsoluteUrl().contains("/api/notifications") && response.getStatus() == 200) {
+                    assertTrue(alicePhone.register(uri, "alice", "1234", aliceContact, 3600, 3600));
+                }
+            }
+        });
+
+        SipCall aliceCall = alicePhone.createSipCall();
+        aliceCall.listenForMessage();
+
+        String from = "+15126002188";
+        String to = "alice";
+        String body = "Hello, Alice!";
+
+        JsonObject callResult = SmsEndpointTool.getInstance().createSms(deploymentUrl.toString(), adminAccountSid,
+                adminAuthToken, from, to, body, null);
+        assertNotNull(callResult);
+
+        assertTrue(aliceCall.waitForMessage(10000));
+        Request messageRequest = aliceCall.getLastReceivedMessageRequest();
+        assertTrue(aliceCall.sendMessageResponse(202, "Accepted", 3600));
+        String messageReceived = new String(messageRequest.getRawContent());
+        assertTrue(messageReceived.equals(body));
+
+        verify(postRequestedFor(urlEqualTo("/api/notifications")));
+    }
+
+    @SuppressWarnings("Duplicates")
+    @Deployment(name = "SmsPushNotificationServerTest", testable = false)
+    public static WebArchive createWebArchiveNoGw() {
+        logger.info("Packaging Test App");
```

Comment:
call reconfigurePorts here.

use new utility and make sure all ports are mapped so we allow parallel testing as in https://github.com/RestComm/Restcomm-Connect/blob/master/restcomm/restcomm.testsuite/src/test/java/org/restcomm/connect/testsuite/sms/SmsTest.java#L577